### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by vietj**
please @purplefox review

### CODE REVIEW

File: src/main/java/io/vertx/ext/httpservicefactory/HttpServiceFactory.java [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+        });
+      } else if (status == 401) {
+        if (prefix().equals("https") && resp.getHeader("WWW-Authenticate") != null && username != null && password != null) {
+          doRequest(client, file, url, username, password, true, unmarshallerFactory, handler);
+          return;
+        }
+        handler.handle(Future.failedFuture(new Exception("Unauthorized")));
+      } else {
+        handler.handle(Future.failedFuture(new Exception("Cannot get file status:" + status)));
+      }
```
Comment (purplefox) [inline src/main/java/io/vertx/ext/httpservicefactory/HttpServiceFactory.java:323]: We should handle redirects too.

I'd recommend looking at the code from Vert.x 2 - you should be able to copy and paste most of it.

File: src/main/java/io/vertx/ext/httpservicefactory/HttpServiceFactory.java [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+import java.util.function.Function;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+/**
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+public class HttpServiceFactory extends ServiceVerticleFactory {
+
```
Comment (purplefox) [inline src/main/java/io/vertx/ext/httpservicefactory/HttpServiceFactory.java:39]: Does it support proxies?

There is code that can be reused from Vert.x 2 there too :)

File: src/main/java/io/vertx/ext/httpservicefactory/package-info.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
+/**
```
Comment (purplefox) [inline src/main/java/io/vertx/ext/httpservicefactory/package-info.java:1]: We should also document the URL format that is assumed, and how to use it for bintray etc.

File: src/main/java/io/vertx/ext/httpservicefactory/HttpServiceFactory.java (hunk 2 above)
Comment (vietj) [inline src/main/java/io/vertx/ext/httpservicefactory/HttpServiceFactory.java:39]: not yet, I will add this to the todo list. I focused rather on signature verification for this first pass.

File: src/main/java/io/vertx/ext/httpservicefactory/package-info.java (hunk 3 above)
Comment (vietj) [inline src/main/java/io/vertx/ext/httpservicefactory/package-info.java:1]: URLs are free form (i.e http/https), there is no special format like in Vert.x 2.

File: src/main/java/io/vertx/ext/httpservicefactory/HttpServiceFactory.java [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+  @Override
+  public String prefix() {
+    return "http";
+  }
+
+  @Override
+  public void resolve(String identifier, DeploymentOptions deploymentOptions, ClassLoader classLoader, Future<String> resolution) {
+
+    int pos = identifier.lastIndexOf("::");
```
Comment (purplefox) [inline src/main/java/io/vertx/ext/httpservicefactory/HttpServiceFactory.java:101]: The docs don't seem to explain anywhere what format is expected... Something containing "::" ??

Comment (vietj) [inline src/main/java/io/vertx/ext/httpservicefactory/HttpServiceFactory.java:101]: good point.

**Pull request merged**